pkg/controller/nodes/subworkflow/handler.go

```diff
@@ -64,26 +65,44 @@ func (w *workflowNodeHandler) Handle(ctx context.Context, nCtx handler.NodeExecu
 	}
 
 	wfNode := nCtx.Node().GetWorkflowNode()
-	workflowPhase := nCtx.NodeStateReader().GetWorkflowNodeState().Phase
+	wfNodeState := nCtx.NodeStateReader().GetWorkflowNodeState()
+	workflowPhase := wfNodeState.Phase
 	if workflowPhase == v1alpha1.WorkflowNodePhaseUndefined {
 		if wfNode == nil {
 			errMsg := "Invoked workflow handler, for a non workflow Node."
 			return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(core.ExecutionError_SYSTEM, errors.RuntimeExecutionError, errMsg, nil)), nil
 		}
 
 		if wfNode.GetSubWorkflowRef() != nil {
-			return updateNodeStateFn(w.subWfHandler.StartSubWorkflow(ctx, nCtx))
+			trns, err := w.subWfHandler.StartSubWorkflow(ctx, nCtx)
+			return updateNodeStateFn(trns, v1alpha1.WorkflowNodePhaseExecuting, err)
 		} else if wfNode.GetLaunchPlanRefID() != nil {
-			return updateNodeStateFn(w.lpHandler.StartLaunchPlan(ctx, nCtx))
+			trns, err := w.lpHandler.StartLaunchPlan(ctx, nCtx)
+			return updateNodeStateFn(trns, v1alpha1.WorkflowNodePhaseExecuting, err)
 		}
 
 		return invalidWFNodeError()
-	}
+	} else if workflowPhase == v1alpha1.WorkflowNodePhaseExecuting {
+		if wfNode.GetSubWorkflowRef() != nil {
+			return w.subWfHandler.CheckSubWorkflowStatus(ctx, nCtx)
+		} else if wfNode.GetLaunchPlanRefID() != nil {
+			return w.lpHandler.CheckLaunchPlanStatus(ctx, nCtx)
+		}
+	} else if workflowPhase == v1alpha1.WorkflowNodePhaseFailing {
```

Comment:
ohh did I already add this phase last time around, thats why I dont see it in this change

Reply:
Yes, it was just not used